### PR TITLE
Adding open source guidelines

### PR DESCRIPTION
**Pull request by wibarre**
**Pull Request check list**

- [ ] Proper tests/regressions included? 
- [ ] Documentation updated?

**Affected functionality**

**Description of change**
Adding open source guidelines for contributions and governance

**Which issue this PR fixes**

### CODE REVIEW

File: GOVERNANCE.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+These are individuals who 1) want to learn more about the Galadriel Project; or 2) are existing users of Galadriel and its tools who wish to follow the Project's progress. 
+
+### Contributors
+
+These are individuals who wish to contribute code or ideas to Galadriel. Contributors submit code and ideas through GitHub.
+
+### Maintainers
+
+These are individuals who can merge submitted PRs into the primary codebase (note: the Project requires PRs be approved by at least one (2) Contributors). For the POC and MVP of the project, contributors and mainteners will have equivalent roles.
```
Comment (mchurichi) [inline GOVERNANCE.md:13]:
```suggestion
These are individuals who can merge submitted PRs into the primary codebase (note: the Project requires PRs to be approved by at least two (2) Contributors). For the POC and MVP of the project, contributors and maintainers will have equivalent roles.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Contributing to the Galadriel Project
+
+The change management process for the Galadriel Project is designed to be transparent, fair, and
+efficient. Anyone may contribute to a project in the galadriel repository that they have read access
+to, provided they:
+
+* Abide by the spiffe [code of conduct](https://github.com/spiffe/spiffe/blob/main/CODE-OF-CONDUCT.md)
```
Comment (mchurichi) [inline CONTRIBUTING.md:7]: nit
```suggestion
* Abide by the SPIFFE [code of conduct](https://github.com/spiffe/spiffe/blob/main/CODE-OF-CONDUCT.md)
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+* Can certify the clauses in the [Developer Certificate of Origin](https://github.com/spiffe/spiffe/blob/main/DCO) 
+
+To get started:
+
+* First, [README](/README.md) to become familiar with how the Galadriel Project is managed
+* Make sure you're familiar with our [Coding Conventions](#conventions) when appropriate
+
+## Sending a pull request
+
+1. Fork the repo
```
Comment (mchurichi) [inline CONTRIBUTING.md:17]: fix numbering

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+1. Commit changes to your fork
+1. Update the docs, if necessary
+1. Ensure your branch is based on the latest commit in `main`
+1. Ensure all tests pass (see project docs for more information)
+2. Make sure your commit messages contain a `Signed-off-by: <your-email-address>` line (see `git-commit --signoff`) to certify the [DCO](/DCO)
+1. Open a [pull request](https://help.github.com/articles/creating-a-pull-request-from-a-fork/)
```
Comment (mchurichi) [inline CONTRIBUTING.md:23]: Add something like...
x. Make sure your all your commits are [GPG-signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  against the upstream `main` branch
+
+All changes to galadriel project must be code reviewed in a pull request (this goes for everyone, even
+those who have merge rights).
+
+## After your pull request is submitted
+
+Pull requests are approved according to the process described in our [governance
+policies](/GOVERNANCE.md). At two other contributors  must approve the pull request.
```
Comment (mchurichi) [inline CONTRIBUTING.md:32]:
```suggestion
policies](/GOVERNANCE.md). At least two other contributors must approve the pull request.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+* Make sure you're familiar with our [Coding Conventions](#conventions) when appropriate
+
+## Sending a pull request
+
+1. Fork the repo
+1. Commit changes to your fork
+1. Update the docs, if necessary
+1. Ensure your branch is based on the latest commit in `main`
+1. Ensure all tests pass (see project docs for more information)
+2. Make sure your commit messages contain a `Signed-off-by: <your-email-address>` line (see `git-commit --signoff`) to certify the [DCO](/DCO)
```
Comment (mchurichi) [inline CONTRIBUTING.md:22]:
```suggestion
2. Make sure all your commit messages contain a `Signed-off-by: <your-email-address>` line (see `git-commit --signoff`) to certify the [DCO](/DCO)
```

**Comment by mchurichi** [comment]
Thanks for adding this, @wibarre!

Could you please rebase/cherry-pick your commits on top of the latest master? seems like there are a few old commits here that are not passing the trivy check

File: GOVERNANCE.md (hunk 1 above)
Comment (wibarre) [reply GOVERNANCE.md:13]: Thanks

File: CONTRIBUTING.md (hunk 2 above)
Comment (wibarre) [reply CONTRIBUTING.md:7]: Always gets me..

File: CONTRIBUTING.md (hunk 4 above)
Comment (wibarre) [reply CONTRIBUTING.md:23]: Agreed.

File: CONTRIBUTING.md (hunk 6 above)
Comment (wibarre) [reply CONTRIBUTING.md:22]: thank you.

File: GOVERNANCE.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+These are individuals who can merge submitted PRs into the primary codebase (note: the Project requires PRs to be approved by at least two (2) Contributors). For the POC and MVP of the project, contributors and maintainers will have equivalent roles.
+
+## Change Review Process
+
+**All changes must be submitted as a GitHub Pull Request (PR)**
+
+The submitter of a PR is responsible for responding to feedback from reviewers and maintainers. While the PR remains open, they are also responsible for ensuring the change is always in a state where it can be merged. Guidelines for submitting a PR for approval can be found [here](/CONTRIBUTING.md).
+
+**All changes must be approved by at least two other Contributors**
```
Comment (mchurichi) [inline GOVERNANCE.md:21]: I know Contributors and Maintainers will be equivalent for the POC, but I'm wondering if we should require the PRs to be approved by Maintainers instead

Comment (wibarre) [reply GOVERNANCE.md:21]: We need to revisit the guidelines once we finish the POC. I was planning on adding a task for that to the backlog. I agree it should be the maintainers in the long run.

Comment (mchurichi) [reply GOVERNANCE.md:21]: Makes sense 👌

**Review by jufantozzi — APPROVED** [review]

**Review by mchurichi — APPROVED** [review]
Lgtm

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+Pull requests are approved according to the process described in our [governance
+policies](/GOVERNANCE.md). At least two other contributors  must approve the pull request.
+
+Once your pull request is submitted, it's your responsibility to:
+
+* Respond to reviewer's feedback
+* Keep it merge-ready at all times until it has been approved and actually merged
+
+Following approval, the pull request will be merged by the submitter of the pull request.
```
Comment (maxlambrecht) [inline CONTRIBUTING.md:40]: Will every submitter have merge permissions? I think it makes sense to have a subset of users with permissions to merge commits into `main`.

**Review by maxlambrecht — APPROVED** [review]

**Pull request closed without merging**